### DISCUSSION TITLE
Question: how to show custom popup dialog for URL

### DISCUSSION THREAD

**Discussion by mvandenabeele** [sibiraj-s/ngx-editor#372] [Q&A] [answered]
Hi,

I'm using ngx-editor for a small project where workers need to be able to insert predefined links. If they need to memorize and retype these URLs, the user experience is not as good as could be, so I'm investigating the possibility to show a popup where they can just select the item they want to link to. Could this be done? Any pointers would be greatly appreciated!

Merijn

**Comment by sibiraj-s** [comment, marked as answer]
You should be able to insert link using commands

```ts
const editor = new Editor()
editor.commands
  .insertLink('Google', {href: 'https://google.com', target: '_blank'})
  .focus()
  .exec()
```

> so I'm investigating the possibility to show a popup

you'll have to create the popup yourself. You can insert the link from there using the above command. There is also  [custom menu](https://sibiraj-s.github.io/ngx-editor/#/menu?id=custom-menu). using custom menu, you can create your own button component which will be rendered at the end of the toolbar item, which you can use to trigger the popup.

**Comment by mvandenabeele** [comment]
So, there's no way to replace the URL dialog with my own dialog?  I'm sorry, still very new to ngx-editor. Thanks for your help. I'll start reading up on custom menu buttons in the meantime.

**Reply by sibiraj-s** [reply]
Yes. you can't replace the URL dialog. It will be straightforward to build own interface instead of replace the builtin popup.

[a comment by an automated account is not shown]
